### ISSUE TITLE
Better Transpiling error reporting

### ISSUE THREAD

**Issue by whitecolor** [stealjs/steal-tools#328]
I've have an error on such case:
There was a module that was actually CJS (it used `require` for importing modules) but inside the code I inaccrately used ES6 style strings using ` character. An it worked in dev mode. But could not build.

It would be nicier if such errors could be indefntified more precisely in thrown error report (at least module where error has been detected).

The error stack, no clue what app's module caused the error, etc:
```
Transpiling...
Potentially unhandled rejection [1] Error: Line 336: Unexpected token ILLEGAL
    at throwError (...\steal-tools\node_modules\transpile\node_modules\esprima\esprima.js:1828:21)
    at scanPunctuator (...\steal-tools\node_modules\transpile\node_modules\esprima\esprima.js:804:9)
    at advance (...\steal-tools\node_modules\transpile\node_modules\esprima\esprima.js:1299:16)
    at lex (...\steal-tools\node_modules\transpile\node_modules\esprima\esprima.js:1342:78)
    at expect (...\steal-tools\node_modules\transpile\node_modules\esprima\esprima.js:1888:21)
    at parseObjectProperty (...\steal-tools\node_modules\transpile\node_modules\esprima\esprima.js:2063:13)
    at parseObjectInitialiser (...\steal-tools\node_modules\transpile\node_modules\esprima\esprima.js:2083:24)
    at parsePrimaryExpression (...\steal-tools\node_modules\transpile\node_modules\esprima\esprima.js:2176:20)
    at parseLeftHandSideExpressionAllowCall (...\steal-tools\node_modules\transpile\node_modules\esprima\esprima.js:2264:61)
    at parsePostfixExpression (...\steal-tools\node_modules\transpile\node_modules\esprima\esprima.js:2317:16)
```

**Comment by matthewp** [comment]
If you can provide a small sample I can create a test, it happens just if you do something like:

```js
require("foo");

`something`
```

?

**Comment by matthewp** [comment]
I was able to recreate it, going to make sure there's a better error here.

**Comment by whitecolor** [comment]
yes right)

**Comment by matthewp** [comment]
So I created #329 which will include the name of the module that failed to transpile. 

Also I'm going to look into seeing if we are using an old version of esprima, it's possible that they support the back-tick strings now.

**Comment by matthewp** [comment]
Ok, I also created https://github.com/stealjs/transpile/pull/49 which will make template strings work, once those tests pass I'll merge and do a new patch release of transpile, then merge #329 and do a patch release of steal-tools. Thanks for reporting.

**Comment by matthewp** [comment]
I've released a transpile 0.9.3 which now works with template strings. Also released a steal-tools 0.12.2 which has better error reporting when there is a transpile error.  Thanks for submitting the issue.

**Comment by matthewp** [comment]
@whitecolor It turns out that Uglify doesn't support template strings currently: https://github.com/mishoo/UglifyJS2/issues/448

Which means this won't work if you plan on minifying our code :-1: So best option if you really need that feature is to use babel/es6.
